module/size-speed-range/foundry/ranges.js

```diff
@@ -0,0 +1,180 @@
+    modifier text for the modifier bucket.
+ */
+export default class GURPSRange {
+    constructor() {
+        this._setup();
+        this._setStrategy(STRATEGY_STANDARD);
+    }
+
+    _setup() {
+        let self = this;
+
+        Hooks.once('init', async function () {
+            game.GURPS = GURPS;
+
+            const choices = {};
+            choices[STRATEGY_STANDARD] = 'Size and Speed/Range Table';
+            choices[STRATEGY_RANGE_BANDS] = 'Monster Hunters 2 Range Bands';
+
+            /* Define Settings */
+            game.settings.register(SYSTEM_NAME, SETTING_NAME, {
+                name: 'Default range modifier strategy:',
+                hint: 'Sets the formula to use to calculate range penalties.',
+                scope: 'world',
+                config: true,
+                type: String,
+                choices,
+                default: STRATEGY_STANDARD,
+                onChange: value => self._updateAfterSettingsChange()
+            });
+        });
+
+        // Set the range to whatever the setting is upon opening the world.
+        // Have to do it this way because the order of "init Hooks" is indeterminate.
+        // This will run after all init hooks have been processed.
+        Hooks.once('ready', async function () {
+            self._updateAfterSettingsChange();
+
+            // Replace the range ruler with our own.
+            Ruler.prototype._getSegmentLabel = (segmentDistance, totalDistance, isTotal) => {
+                const units = canvas.scene.data.gridUnits;
+                const getLabel = (distance, units) => `${Math.round(distance * 100) / 100} ${units}`;
+
+                let label = getLabel(segmentDistance, units);
+                if (isTotal && segmentDistance !== totalDistance) {
+                    label += ` [${getLabel(totalDistance, units)}]`;
+                }
+
+                const mod = self._getRangePenalty(totalDistance, units);
+                game.GURPS.ModifierBucket.setTempRangeMod(mod);
+
+                return label + ` (${mod})`;
+            }
+
+            Ruler.prototype._endMeasurementOrig = Ruler.prototype._endMeasurement;
+            Ruler.prototype._endMeasurement = function () {
+                this._endMeasurementOrig();
+                game.GURPS.ModifierBucket.addTempRangeMod();
+            }
+        })
+    }
+
+    _getRangePenalty(distance, units) {
+        if (getStrategy() === STRATEGY_RANGE_BANDS) {
+            return RangeBandTable.yardsToPenalty(
+                convertToYards({
+                    value: distance,
+                    inputUnit: parseDistance(`${distance} ${units}`).unit,
+                    conversionMethod: conversionMethods.real
+                })
+            );
+        }
+
+        return SSRT.speedRangeFromExpression({expression: `${distance} ${units}`});
+    }
+
+    async _updateAfterSettingsChange() {
+        const currentStrategy = getStrategy();
```

Comment:
variables should always be named to accurately express their purpose; currentStrategy is much better than currentValue in that regard